Installation Error

Hi,
i am getting an error while installing the plugin as root:

root@raspberrypi:~# npm install -g homebridge-xiaomi-roborock-vacuum@latest --unsafe-perm

> deasync@0.1.19 install /usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/deasync
> node ./build.js

make: Verzeichnis „/usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/deasync/build“ wird betreten
  CXX(target) Release/obj.target/deasync/src/deasync.o
make: g++: Kommando nicht gefunden
deasync.target.mk:107: die Regel für Ziel „Release/obj.target/deasync/src/deasync.o“ scheiterte
make: *** [Release/obj.target/deasync/src/deasync.o] Fehler 127
make: Verzeichnis „/usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/deasync/build“ wird verlassen
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 4.19.66-v7+
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/deasync
gyp ERR! node -v v12.16.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
Build failed
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! deasync@0.1.19 install: `node ./build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the deasync@0.1.19 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-15T16_46_17_998Z-debug.log

Any idea how to solve this?

Greetz,
Doc

--- a/index.js
+++ b/index.js
@@ -5,7 +5,12 @@ const miio = require('miio-nicoh88');
 const util = require('util');
 const callbackify = require('./lib/callbackify');
 const safeCall = require('./lib/safeCall');
-const sleep = require('system-sleep');
+let sleep;
+try {
+  sleep = require('system-sleep');
+} catch (e) {
+  // noop
+}
 
 let homebrideAPI, Service, Characteristic;
 
@@ -898,10 +903,17 @@ class XiaomiRoborockVacuum {
   }
 
   getServices() {
-    if (this.config.delay)
-      sleep(5000);
+    if (this.config.delay) this.sleep(5000);
     this.log.debug(`DEB getServices | ${this.model}`);
     return Object.keys(this.services).map((key) => this.services[key]);
+  }
+
+  sleep(time) {
+    if (sleep) {
+      sleep(time);
+    } else {
+      this.log.warn(`Can't use the delay option because sleep failed to install`);
+    }
   }
 
   // CONSUMABLE / CARE

--- a/index.js
+++ b/index.js
@@ -912,7 +912,8 @@ class XiaomiRoborockVacuum {
     if (sleep) {
       sleep(time);
     } else {
-      this.log.warn(`Can't use the delay option because sleep failed to install`);
+      this.log.warn(`Can't use the delay option because the module "system-sleep" failed to install.\n
+      Make sure this optional dependency is properly installed if you want to use the "delay" option.`);
     }
   }
 

--- a/package.json
+++ b/package.json
@@ -36,7 +36,9 @@
   },
   "dependencies": {
     "miio-nicoh88": ">=0.1.0",
-    "semver": "^7.1.2",
+    "semver": "^7.1.2"
+  },
+  "optionalDependencies": {
     "system-sleep": "^1.3.6"
   }
 }

